spinta/types/model.py

```diff
@@ -381,11 +384,30 @@ def check(context: Context, model: Model):
 
 @check.register(Context, Property)
 def check(context: Context, prop: Property):
+    commands.check(context, prop.dtype)
     if prop.enum:
         for value, item in prop.enum.items():
             commands.check(context, item, prop.dtype, item.prepare)
 
 
+@check.register(Context, DataType)
+def check(context: Context, dtype: DataType):
+    pass
+
+
+@check.register(Context, Ref)
+def check(context: Context, dtype: Ref):
+    if dtype.prop.level:
+        level = int(dtype.prop.level)
+        ref_external = dtype.model.external
+        external = dtype.prop.model.external
+        if ref_external and external and \
+                ref_external.resource and \
+                ref_external.resource != external.resource and \
+                level > 3:
+            raise exceptions.InvalidPropertyLevel(dtype.prop.model, prop=dtype.prop.name)
```

Comment:
```suggestion
            raise exceptions.InvalidPropertyLevel(dtype)
```

Exceptions should always use deepest node possible, higher nodes will be populated to error context automagically.